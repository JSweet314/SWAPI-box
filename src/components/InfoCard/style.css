.info-card {
  height: 60%;
  margin: 1rem;
  padding: 1rem;
  width: 40%;
  border: 1px solid black;
}

.info-card p {
  font-size: 1.5rem;
}

.info-card__top-line {
  display: flex;
  font-size: 2rem;
  justify-content: space-between;
  border-bottom: 1px dashed lightgray;
  padding: 0 0 1rem 0;
}

.info-card button {
  height: 4rem;
  width: 4rem;
  background-color: gray;
  background-image: url('../../images/star.svg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: 2rem 2rem;
  border: none;
}

.info-card .selected {
  background-image: url('../../images/star-active.svg');
}