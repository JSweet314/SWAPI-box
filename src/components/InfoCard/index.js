import React from 'react';
import './style.css';
import PropTypes from 'prop-types';

const InfoCard = ({ name, homeworld, species, population }) => {
  return (
    <article className="info-card">
      <h3>{name}</h3>
      <p>{homeworld}</p>
      <p>{species}</p>
      <p>{population}</p>
    </article>
  );
};

InfoCard.propTypes = {
  name: PropTypes.string.isRequired,
  homeworld: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  population: PropTypes.string.isRequired
};

export default InfoCard;