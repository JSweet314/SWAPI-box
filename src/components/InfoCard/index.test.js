import React from 'react';
import { shallow } from 'enzyme';
import InfoCard from './index';

describe('InfoCard', () => {
  const card = {
    homeworld: 'tatooine',
    species: 'human',
    name: 'Obi-Wan',
    population: 1000
  };

  it('should match a snapshot', () => {
    const wrapper = shallow(<InfoCard card={card}/>, {disableLifecycleMethods: true});

    expect(wrapper).toMatchSnapshot();
  });
});