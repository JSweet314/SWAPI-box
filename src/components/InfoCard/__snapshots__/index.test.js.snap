// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoCard
    card={
        Object {
            "homeworld": "tatooine",
            "name": "Obi-Wan",
            "population": 1000,
            "species": "human",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": false,
      "className": "info-card",
    },
    "ref": null,
    "rendered": false,
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": false,
        "className": "info-card",
      },
      "ref": null,
      "rendered": false,
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
